packages/api/src/Client.ts

```diff
@@ -323,20 +338,31 @@ export default class Client extends EventEmitter {
         let errorMessage = message.data.error;
 
         if (errorMessage == '13') {
-          errorMessage = '[Error 13] Permission denied. You are trying to access a file/directory without having the necessary permissions. Most likely one of the plot folders in your config.yaml has an issue.';
+          errorMessage =
+            '[Error 13] Permission denied. You are trying to access a file/directory without having the necessary permissions. Most likely one of the plot folders in your config.yaml has an issue.';
         } else if (errorMessage == '22') {
-          errorMessage = '[Error 22] File not found. Most likely one of the plot folders in your config.yaml has an issue.';
+          errorMessage =
+            '[Error 22] File not found. Most likely one of the plot folders in your config.yaml has an issue.';
         }
 
         log(`Request ${requestId} rejected`, errorMessage);
 
+        if (message?.data?.errorDetails?.message) {
```

Comment:
Maybe we should use "else if" instead just "if".
Second option is to use this message directly on line 338.

Reply:
Moved it into an else if grouped with the logic right before. The error message is set in this case, but there's also error details. So the idea was to combine both into a single message to the user. This form of error message is only used by the keyring backend code currently.